### PR TITLE
docs(api): add docs for `<NuxtPage>` component

### PR DESCRIPTION
**Pull request by Krutie**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Added API documentation for `NuxtPage` component page

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/3.api/2.components/1.nuxt-page.md [hunk 1]
```diff
@@ -1,7 +1,55 @@
...
+
+`NuxtPage` is a wrapper around `<RouterView>` component from Vue Router. As a result, `NuxtPage` component accepts `name` and `route` props. 
+
+- **name:** type: `string`
+
+`name` helps `RouterView` render the component with the corresponding name in the matched route record's components option.
+
+- **route:** type: `RouteLocationNormalized`
+
+`route` is a route location that has all of its component resolved.
```
Comment (dammy001) [inline docs/content/3.api/2.components/1.nuxt-page.md:13]:
```suggestion
`route` is a route location that has all of its components resolved.
```

File: docs/content/3.api/2.components/1.nuxt-page.md [hunk 2]
```diff
@@ -1,7 +1,55 @@
...
+
+> **Nuxt automatically resolves the `name` and `route` by scanning and rendering all Vue component files found in the `/pages` directory.**
+
+Apart from the `name` and `route`, `NuxtPage` component also accepts `pageKey` props.
+
+- **pageKey:** type: `string` or `function`
+
+`pageKey` helps you control when the `NuxtPage` component is re-rendered.
```
Comment (dammy001) [inline docs/content/3.api/2.components/1.nuxt-page.md:21]:
```suggestion
`pageKey` helps control when the `NuxtPage` component is re-rendered.
```

File: docs/content/3.api/2.components/1.nuxt-page.md [hunk 3]
```diff
@@ -1,7 +1,55 @@
...
+  key: route => route.fullPath
+})
+</script>
+```
+
+:button-link[Open on StackBlitz]{href="https://stackblitz.com/github/nuxt/framework/tree/main/examples/routing/pages?file=app.vue" blank}
+
+## Custom props
+
+In addition, `NuxtPage` also accepts custom props that you may need to pass further down the hierarchy. These custom props are accessible via `attrs` in Nuxt app.
```
Comment (dammy001) [inline docs/content/3.api/2.components/1.nuxt-page.md:46]:
```suggestion
In addition, `NuxtPage` also accepts custom props that you may need to pass further down the hierarchy. These custom props are accessible via `attrs` in the Nuxt app.
```

File: docs/content/3.api/2.components/1.nuxt-page.md [hunk 4]
```diff
@@ -1,7 +1,55 @@
 # `<NuxtPage>`
 
-::ReadMore{link="/guide/directory-structure/app"}
-::
+ `<NuxtPage>` is an in-built component that comes with Nuxt. `NuxtPage` component is required to display top-level or nested pages located in the `/pages` directory.
```
Comment (pi0) [inline docs/content/3.api/2.components/1.nuxt-page.md:3]:
```suggestion
 `<NuxtPage>` is a built-in component that comes with Nuxt. `NuxtPage` component is required to display top-level or nested pages located in the `/pages` directory.
```

File: docs/content/3.api/2.components/1.nuxt-page.md [hunk 5]
```diff
@@ -1,7 +1,55 @@
...
+
+`NuxtPage` is a wrapper around `<RouterView>` component from Vue Router. As a result, `NuxtPage` component accepts `name` and `route` props.
```
Comment (pi0) [inline docs/content/3.api/2.components/1.nuxt-page.md:5]:
```suggestion
`NuxtPage` is a wrapper around [`<RouterView>`](https://router.vuejs.org/api/#router-view-props) component from Vue Router. `NuxtPage` component accepts same `name` and `route` props. 
```

**Review by pi0 — APPROVED** [review]
Really nice addition! Thanks for working in this <3

**Pull request merged**